app/src/main/java/com/framgia/wsm/screen/requestleave/listrequestleave/ListRequestLeaveViewModel.java

```diff
@@ -1,25 +1,123 @@
 
-    public ListRequestLeaveViewModel(ListRequestLeaveContract.Presenter presenter) {
+    public ListRequestLeaveViewModel(Context context, ListRequestLeaveContract.Presenter presenter,
+            DialogManager dialogManager) {
+        mContext = context;
         mPresenter = presenter;
         mPresenter.setViewModel(this);
+        mDialogManager = dialogManager;
+        mCurrentPositionStatus = PositionStatus.ALL;
+        mCurrentStatus = mContext.getString(R.string.all);
+        mCalendar = Calendar.getInstance();
+        mYear = mCalendar.get(Calendar.YEAR);
+        mMonth = mCalendar.get(Calendar.MONTH);
+        mDialogManager.dialogMonthYearPicker(this, mYear, mMonth);
     }
 
     @Override
     public void onStart() {
         mPresenter.onStart();
     }
 
+    public void onPickTypeStatus(View view) {
+        mDialogManager.dialogListSingleChoice(mContext.getString(R.string.status), R.array.status,
+                mCurrentPositionStatus, new MaterialDialog.ListCallbackSingleChoice() {
+                    @Override
+                    public boolean onSelection(MaterialDialog materialDialog, View view,
+                            int positionType, CharSequence charSequence) {
+                        setCurrentPositionStatus(positionType);
+                        setCurrentStatus(String.valueOf(charSequence));
+                        return true;
+                    }
+                });
+    }
+
+    public void onPickMonthYear(View view) {
+        mDialogManager.showMonthYearPickerDialog();
+    }
+
+    public void setCurrentPositionStatus(int currentPositionStatus) {
+        mCurrentPositionStatus = currentPositionStatus;
+    }
+
+    @Bindable
+    public String getCurrentStatus() {
+        return mCurrentStatus;
+    }
+
+    public void setCurrentStatus(String currentStatus) {
+        mCurrentStatus = currentStatus;
+        notifyPropertyChanged(BR.currentStatus);
+    }
+
+    @Bindable
+    public String getMonthYear() {
+        return mMonthYear;
+    }
+
+    public void setMonthYear(String monthYear) {
+        mMonthYear = monthYear;
+        notifyPropertyChanged(BR.monthYear);
+    }
+
     @Override
     public void onStop() {
         mPresenter.onStop();
     }
+
+    @Override
+    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
+        mYear = year;
+        mMonth = month;
+        updateMonthYear();
+    }
+
+    private void updateMonthYear() {
+        int localMonth = (mMonth + 1);
+        String month = localMonth < 10 ? "0" + localMonth : Integer.toString(localMonth);
+        String year = Integer.toString(mYear);
+        setMonthYear(String.valueOf(
+                new StringBuilder().append(month).append("/").append(year).append("")));
+    }
+
+    @IntDef({
+            PositionStatus.ALL, PositionStatus.ACCEPTED, PositionStatus.PENDING,
+            PositionStatus.REJECTED
+    })
+    @interface PositionStatus {
+        int ALL = 0;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.